macOS/DuckDuckGo/Suggestions/Model/SuggestionContainer.swift

```diff
@@ -33,8 +33,17 @@ final class SuggestionContainer {
 
     typealias OpenTabsProvider = @MainActor () -> [any Suggestions.BrowserTab]
     private let openTabsProvider: OpenTabsProvider
-    private let historyCoordinating: HistoryCoordinating
-    private let bookmarkManager: BookmarkManager
+
+    protocol HistoryProvider {
+        func history(for suggestionLoading: SuggestionLoading) -> [HistorySuggestion]
+    }
+    private let historyProvider: HistoryProvider
+
+    protocol BookmarkProvider {
+        func bookmarks(for suggestionLoading: SuggestionLoading) -> [Suggestions.Bookmark]
+    }
+    private let bookmarkProvider: BookmarkProvider
```

Comment:
Detach the Suggestion Container from `BookmarkManager` and `HistoryCoordinator` implementations